aten/src/ATen/EmptyTensor.cpp

```diff
@@ -275,6 +275,71 @@ TensorBase empty_meta(
   return empty_meta(size, dtype, memory_format_opt);
 }
 
+TensorBase empty_symint_meta(
+  SymIntArrayRef size,
+  c10::optional<ScalarType> dtype_opt,
+  c10::optional<Layout> layout_opt,
+  c10::optional<Device> device_opt,
+  c10::optional<bool> pin_memory_opt,
+  c10::optional<c10::MemoryFormat> memory_format_opt
+) {
+  auto device = device_or_default(device_opt);
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(device.type() == DeviceType::Meta);
+  // NB: because there is no SparseMeta (yet), non-strided layout is
+  // exerciseable
+  TORCH_CHECK_NOT_IMPLEMENTED(
+    layout_or_default(layout_opt) == Layout::Strided,
+    "non-strided meta tensors not supported yet"
+  );
+
+  auto scalar_type = dtype_or_default(dtype_opt);
+  auto *allocator = GetAllocator(kMeta);
+  constexpr c10::DispatchKeySet meta_dks(c10::DispatchKey::Meta);
+  // TODO: do this.  Note that naive implementation will choke on truly
+  // unknown sizes without on the fly reasoning
+  // at::detail::check_size_nonnegative(size);
+  at::detail::raise_warning_for_complex_half(scalar_type);
+  caffe2::TypeMeta dtype = scalarTypeToTypeMeta(scalar_type);
+  SymInt size_bytes = dtype.itemsize();
+  for (auto s : size) {
+    size_bytes = size_bytes * s;
+  }
+  auto storage_impl = c10::make_intrusive<StorageImpl>(
+      c10::StorageImpl::use_byte_size_t(),
+      size_bytes,
+      allocator,
+      /*resizeable=*/true);
+
+  auto tensor = detail::make_tensor_base<TensorImpl>(
+      std::move(storage_impl), meta_dks, dtype);
+
+  int64_t dim = size.size();
+  std::vector<SymInt> strides;
+  strides.resize(dim);
+
+  // TODO: Move this into TensorImpl
```

Comment:
Isn't that already? You have the `empty_tensor_restride(MemoryFormat::Contiguous)` that does this.

Reply:
nit: it does feel like we need a helper here.

Reply:
`empty_tensor_restride` doesn't operate on symbolic integers. I'm a little torn about whether or not to directly SymInt'ify it or maintain two separate versions; there's a lot of arithmetic operations going on and it would be better for them to be all unchecked